components/automate-cli/pkg/verifyserver/server/api/v1/externalpostgresql_test.go

```diff
@@ -0,0 +1,117 @@
+package v1_test
+
+import (
+	"io/ioutil"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/models"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/server"
+	v1 "github.com/chef/automate/components/automate-cli/pkg/verifyserver/server/api/v1"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/services/externalpostgresqlservice"
+	"github.com/chef/automate/components/automate-cli/pkg/verifyserver/utils/fiberutils"
+	"github.com/chef/automate/lib/logger"
+	"github.com/gofiber/fiber"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	ExternalPgSuccessConnectionTitle  = "Connection successfully tested"
+	ExternalPgFailConnectionTitle     = "External Postgresql Connection failed"
+	ExternalPgConnectionErrorMsg      = "Machine is unable to connect with External Managed Postgresql"
+	ExternalPgConnectionResolutionMsg = "Ensure that the Postgres configuration provided is correct"
+	ExternalPgConnectionSuccessMsg    = "Connection successfully tested"
+	ExternalPgDebugMsg                = "Review security group or firewall settings as well on the infrastructure"
+	ExternalPgsuccessMessage          = "{\"status\":\"SUCCESS\",\"result\":{\"passed\":true,\"checks\":[{\"title\":\"Connection successfully tested\",\"passed\":true,\"status\":\"\",\"success_msg\":\"Connection successfully tested\",\"error_msg\":\"\",\"resolution_msg\":\"\",\"debug_msg\":\"\"}]}}"
+	ExternalPgfailureMessage          = "{\"status\":\"SUCCESS\",\"result\":{\"passed\":false,\"checks\":[{\"title\":\"External Postgresql Connection failed\",\"passed\":false,\"status\":\"\",\"success_msg\":\"\",\"error_msg\":\"Machine is unable to connect with External Managed Postgresql\",\"resolution_msg\":\"Ensure that the Postgres configuration provided is correct\",\"debug_msg\":\"Review security group or firewall settings as well on the infrastructure\"}]}}"
+	errorBodyParser                   = "{\"status\":\"FAILED\",\"result\":null,\"error\":{\"code\":400,\"message\":\"invalid character '}' looking for beginning of object key string\"}}"
+)
+
+func SetupMockExternalPostgresqlService() externalpostgresqlservice.ISExternalPostgresqlService {
+	return &externalpostgresqlservice.MockExternalPostgresqlService{
+		GetPgConnectionFunc: func(*models.ExternalPgRequest) (*models.ExternalPgResponse, error) {
```

Comment:
I think you need to add a test where GetPgConnection function returns error. Currently I can see only success mock response

Reply:
As i have used sql libraries i cannot handle errors in unit tests

Reply:
As discussed this has a test case now